### PR TITLE
feat(api-server): return result set for the new query api

### LINKED ISSUE #2127: feat(api-server): return result set for the new query api
**Describe the feature you'd like**

This is the enhancement for #2091 

The interface `POST http://127.0.0.1:9080/dbs/<db_name>` accepts a query query from request body and returns query status, but no query results. 

## solution
The `ResultSet` returned from `sql_router_->ExecuteSQL` should encoded into http response buffer. Impl logic should similar to procedure call results in https://github.com/4paradigm/OpenMLDB/blob/053c4e3d885b237711d3719be7bf0c1905c5197e/src/apiserver/api_server_impl.cc#L385-L394

**Additional context**

### PR DESCRIPTION
**Pull request by qsliu2017**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

close #2127 

* **What is the current behavior?** (You can also link to an open issue here)

`POST /dbs/:db_name` only return "code" and "msg"

* **What is the new behavior (if this is a feature change)?**

return query result like
```json
{ "code": 0, "msg": "ok", "data": [[1, "bb"]] }
```

### CODE REVIEW

**Comment by aceforeverd** [comment]
api_server_test failed ?

**Review by dl239 — APPROVED** [review]

**Comment by qsliu2017** [comment]
> api_server_test failed ?

In the added test case, I try to send a query to API server, but [it turns out `Fail to get TaskManager client`](https://github.com/4paradigm/OpenMLDB/runs/7462829109?check_suite_focus=true#:~:text=I0722%2006%3A13%3A56.456756,%5B%20RUN%20%20%20%20%20%20%5D%20APIServerTest.invalidPut).

Seems like the API server test env do not have a real database? need helps @aceforeverd

**Comment by aceforeverd** [comment]
> > api_server_test failed ?
> 
> In the added test case, I try to send a query to API server, but [it turns out `Fail to get TaskManager client`](https://github.com/4paradigm/OpenMLDB/runs/7462829109?check_suite_focus=true#:~:text=I0722%2006%3A13%3A56.456756,%5B%20RUN%20%20%20%20%20%20%5D%20APIServerTest.invalidPut).
> 
> Seems like the API server test env do not have a real database? need helps @aceforeverd

I'll check today

**Comment by aceforeverd** [comment]
[ospp - Golang SDK](https://trello.com/c/7aquDamb/118-ospp-golang-sdk)

File: src/apiserver/api_server_test.cc [hunk 1]
```diff
@@ -145,6 +145,45 @@ TEST_F(APIServerTest, jsonFormat) {
...
+    std::string insert_sql = "insert into demo values (1, \"bb\");";
+    ASSERT_TRUE(env->cluster_sdk->Refresh());
+    ASSERT_TRUE(env->cluster_remote->ExecuteInsert(env->db, insert_sql, &status));
+
+    {
+        brpc::Controller cntl;
+        cntl.http_request().set_method(brpc::HTTP_METHOD_POST);
+        cntl.http_request().uri() = "http://127.0.0.1:8010/dbs/" + env->db;
+        cntl.request_attachment().append(R"({
+            "sql": "select c1, c2 from demo;", "mode": "offsync"
```
Comment (aceforeverd) [inline src/apiserver/api_server_test.cc]: this require ` "mode": "online"` field, api server default to `offline` mode

**Review by aceforeverd — CHANGES_REQUESTED** [review]

File: src/apiserver/api_server_test.cc (hunk 1 above)
Comment (qsliu2017) [reply src/apiserver/api_server_test.cc]: addressed

**Review by vagetablechicken — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**